src/material.cpp

```diff
@@ -221,19 +260,39 @@ void pyne::Material::from_hdf5(std::string filename, std::string datapath, int r
   // Open the database
   hid_t db = H5Fopen(filename.c_str(), H5F_ACC_RDONLY, fapl);
 
-  bool datapath_exists = h5wrap::path_exists(db, datapath);
-  if (!datapath_exists)
-    throw h5wrap::PathNotFound(filename, datapath);
 
   // Clear current content
   comp.clear();
-
   // Load via various protocols
-  if (protocol == 0)
+  if (protocol == 0) {
+    bool datapath_exists = h5wrap::path_exists(db, datapath);
+    if (!datapath_exists)
+      throw h5wrap::PathNotFound(filename, datapath);
     _load_comp_protocol0(db, datapath, row);
-  else if (protocol == 1)
-    _load_comp_protocol1(db, datapath, row);
-  else
+  } else if (protocol == 1) {
+    enum prot1_layout {path_donotexists=-1, unknown, old_layout, new_layout};
```

Comment:
Define this as a private member of the `Material` class and then use them as the return types of the function?